sdk/src/types/block/output/account.rs

```diff
@@ -239,7 +238,8 @@ impl AccountOutputBuilder {
             AccountOutput::KIND,
             features.native_token(),
             self.mana,
-        )?;
+        )
+        .map_err(UnlockConditionError::from)?;
```

Comment:
Tbh, even though it mainly return address errors, `verify_restricted_addresses` is an UnlockCondition function, I'd have it return `UnlockConditionError` directly and remove the maps. But not super opinionated, feel free to resolve.

Reply:
alrighty